### PR TITLE
Small updates to wkf tools

### PR DESCRIPTION
**Pull request by kirstybayliss**
This PR implements the following small updates to the wkf:
- adds a tool to check a toml file contains all necessary details, make a working copy, and set zones to use from a list
- updates set_defaults so that defaults can be added only for a list of specified zones
- set_gr_params now checks lists properly
- adding new plotting options for focal mechanism data (this should maybe be in a different location?)
- create_subcatalogues takes a given list of sources and passes on all catalogue info
- analysis_nodal_plane now passes on the option to save an output csv with a classification column
- added the option to only pass selected zones (by id) to wkf functions
- mfd check now works with multimfd of truncGutenbergRichterMFD type
- updates h3_polyfill to use 'geojson_compliant' parameter, meaning we don't need to switch the lat/lon columns
- updates the ghm file to get new ocean island site locations
- adds code for sanity-checking source xmls (combined from various notebooks)

### CODE REVIEW

File: openquake/ghm/mosaic.py [hunk 1]
```diff
@@ -39,6 +39,22 @@
            'nea': {'RUS': ['75 0 180 0 180 88 75 88',
                            '-180 0 -160 0 -160 88 -180 88']},
            'nwa': {'RUS': ['75 0 25 0 25 10 25 20 25 40 25 50 25 88 75 88']},
+           'oan': {'ATF': ['49.5 -50.1 71 -50.1 71 -45.6 49.5 -45.6']},
```
Comment (kejohnso) [inline openquake/ghm/mosaic.py:42]: do we have a model 'oan'? (Not sure if it matters code-wise... )

File: openquake/mbi/wkf/create_smoothing_per_zone.py [hunk 2]
```diff
@@ -5,9 +5,9 @@
 from openquake.wkf.seismicity.smoothing import create_smoothing_per_zone
 
 
-def main(fname_points: str, fname_polygons: str, folder_out: str='/tmp',
-         skip=[]):
-    create_smoothing_per_zone(fname_points, fname_polygons, folder_out, skip)
+def main(fname_points: str, fname_polygons: str, folder_out: str='/tmp',*,
+         skip=[], use = []):
+    create_smoothing_per_zone(fname_points, fname_polygons, folder_out, skip, use)
```
[the remark this answers is not in the document]
Comment (kejohnso) [reply openquake/mbi/wkf/create_smoothing_per_zone.py:10]: I believe we decided this is correct as is

File: openquake/mbi/wkf/set_defaults.py [hunk 3]
```diff
@@ -7,8 +7,13 @@
 from openquake.wkf.utils import get_list
 
 
-def main(fname_conf: str, fname_defaults: str):
+def main(fname_conf: str, fname_defaults: str,  use: str = [], skip: str = []):
```
[the remark this answers is not in the document]
Comment (kejohnso) [reply openquake/mbi/wkf/set_defaults.py:10]: also decided this is okay...

File: openquake/mbi/wkf/set_h3_to_zones.py [hunk 4]
```diff
@@ -5,12 +5,12 @@
 from openquake.wkf.h3.zones import discretize_zones_with_h3_grid
 
 
-def main(h3_level: str, fname_poly: str, folder_out: str):
+def main(h3_level: str, fname_poly: str, folder_out: str,*, use: str = []):
```
[the remark this answers is not in the document]
Comment (kejohnso) [reply openquake/mbi/wkf/set_h3_to_zones.py:9]: as above (OK)

File: openquake/wkf/mfd.py [hunk 5]
```diff
@@ -35,6 +35,7 @@
 from openquake.mbt.tools.mfd import (
     EEvenlyDiscretizedMFD,
     get_evenlyDiscretizedMFD_from_truncatedGRMFD)
+from openquake.hazardlib.mfd.multi_mfd import MultiMFD
```
[the remark this answers is not in the document]
Comment (kejohnso) [reply openquake/wkf/mfd.py:38]: @kirstybayliss pls double check this

[the remark this answers is not in the document]
Comment (kirstybayliss) [reply openquake/wkf/mfd.py:38]: yep this is completely unnecessary and I don't remember why I imported it before. Removed.

File: openquake/ghm/mosaic.py (hunk 1 above)
Comment (kirstybayliss) [reply openquake/ghm/mosaic.py:42]: whoops this should be 'pan' (polar Antarctic)

**Review by kejohnso — APPROVED** [review]

**Pull request merged**